quarto/03-QC-LBT01.qmd

```diff
@@ -0,0 +1,172 @@
+                               list(everything() ~ continuous_summary_fns(c("N", "mean", "sd", "median", "min", "max"))))
+
+# calculate overall statistics
+ard_overall <- ard_continuous(adlb, 
+                             by = c(PARAMCD, AVISIT), # note arm var is removed
+                             variables = c(AVAL, CHG), 
+                             statistic = 
+                               list(everything() ~ continuous_summary_fns(c("N", "mean", "sd", "median", "min", "max"))))
+```
+
+### 4.Statistics comparison
+
+#### A. {rtables} reformat
+Some reformatting is required to ensure the resulting data.frames are compatible. Most of the reformatting is adjusting the column names to match that of the ARD result.
+
+  1.  rename `group1` and `group1_level`, `ARM` and `ARM` levels accordingly.
+  2.  rename `group2` and `group2_level`, `PARAMCD` and `PARAMCD` levels accordingly.
+  3.  rename `group3` and `group3_level`, `AVISIT` and `AVISIT` levels accordingly.
+  4.  recode `stat` `n` value to `N` to match ARD
+  5.  Select columns relevant to comparison
+
+We'll also separate the statistics related to the overall ARM to match the `ard_overall` object.
+```{r}
+#| code-fold: false
+#| code-summary: "Show the code"
+rtables_result <- rtables_result |>
+  dplyr::mutate(stat_name = dplyr::recode(stat_name, "n" = "N"),
+                group4_level = dplyr::recode(group4_level, 
+                                             "Analysis Value" = "AVAL",
+                                             "Absolute Change from Baseline" = "CHG")) |>
+  dplyr::rename(group1 = group3,
+                group1_level = group3_level,
+                group2 = group1,
+                group2_level = group1_level,
+                group3 = group2,
+                group3_level = group2_level,
+                variable = group4_level,
+                # giving a dummy name for this column so it doesn't interfere with the column renaming
+                val = variable) |>
+  dplyr::select(c(group1, group1_level, group2, group2_level, group3, group3_level, variable, stat_name, stat))
+
+rtables_result_overall <- rtables_result |>
+  dplyr::filter(group1_level == "Overall ARM")
+
+# remove the overall observations from the results df
+rtables_result <- rtables_result |>
+  dplyr::filter(group1_level != "Overall ARM")
+```
+Based on dimensions alone - we can see that tables likely generated similar results
+```{r, eval=TRUE}
+print("rtables Dimensions")
+dim(rtables_result)
+print("ARD Dimensions")
+dim(ard_result)
+```
+
+##### B. ARD reformat
+We'll also need some format adjustments for the ARD object. Primarily, converting lists to vectors (character or numeric) to match the rtables_result format. Not doing so will result in an incompatible comparison errors in `diffdf()`.
+
+1.  Select columns relevant to comparison
+2.  Convert any list columns to character and numeric
+3.  Convert any `NaN` stats to `NA` to match {rtables}
```

Comment:
```suggestion
3.  Convert any `NaN` stats to `NA` to match {rtables}

```

Reply:
Should there be a blank line before a code block? I want to make sure it's consistent throughout all the documents.